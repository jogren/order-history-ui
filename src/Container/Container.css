.Container_section {
  background-color: #CECECE;
}