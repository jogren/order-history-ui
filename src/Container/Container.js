import React from 'react';
import Order from '../Order/Order';
import './Container.css';

const Container = ({ orders, deletePurchase }) => {
  let ordersList = orders.map(order => {
    return <Order key={order.id} id={order.id} description={order.description} img={order.img} name={order.name} price={order.price} deletePurchase={deletePurchase}/>
  })
  return (
    <section className="Container_section">
      {ordersList}
    </section>
  )

}

export default Container;