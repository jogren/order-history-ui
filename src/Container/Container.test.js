import React from 'react';
import { shallow } from 'enzyme';
import Container from './Container';

describe('Container', () => {
  let wrapper;
  let mockOrder;

  beforeEach(() => {
    mockOrder = {
      id: 1,
      name: 'iPhone',
      description: 'It\'s the new iPhone 18',
      price: 800,
      img: 'https://store.storeimages.cdn-apple.com/4982/as-images.apple.com/is/iphone-xs-max-gold-select-2018?wid=940&hei=1112&fmt=png-alpha&qlt=80&.v=1550795409154'
    }

    wrapper = shallow(<Container orders={[mockOrder]} deletePurchase={jest.fn()} />)
  })

  it('should keep an up-to-date snapshot of Container', () => {
    expect(wrapper).toMatchSnapshot();
  })
})