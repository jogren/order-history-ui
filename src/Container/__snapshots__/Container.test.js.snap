// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should keep an up-to-date snapshot of Container 1`] = `
<section
  className="Container_section"
>
  <Order
    deletePurchase={[MockFunction]}
    description="It's the new iPhone 18"
    id={1}
    img="https://store.storeimages.cdn-apple.com/4982/as-images.apple.com/is/iphone-xs-max-gold-select-2018?wid=940&hei=1112&fmt=png-alpha&qlt=80&.v=1550795409154"
    key="1"
    name="iPhone"
    price={800}
  />
</section>
`;
